src/Telemetry/AzureDevOpsClient/AzureDevOpsClient.cs

```diff
@@ -155,6 +162,40 @@ public async Task<WorkItem> CreateRcaWorkItem(string project, string title, Canc
             return JsonConvert.DeserializeObject<WorkItem>(json);
         }
 
+        public async Task<string> TryGetLogContents(string logUri, CancellationToken cancellationToken)
+        {
+            for (int i = 0; i < _retryNumber; i++)
```

Comment:
Microsoft.DotNet.Services.Utility has ExponentialRetry

Reply:
I don't think this would work for us, because we need to use the base delay that AzDo is sending us with the response, in case of throttling. I'm not seeing any mechanism in this class that would let us include this delay, in addition to the random delay

Reply:
Can we extend this class to account for that? We probably want to check the header and behave this way for any retry we use it for

Reply:
We could, but I'm not sure if it belongs in this class, from what I'm seeing, this is meant to be a generic Retry class that can be used in any case, this would be specific to Http stuff

Reply:
Ah, understood

Reply:
See my other comment about using the RetryAfterHandler from helix and not needing to deal with this at all.

Reply:
If you combine the RetryAfterHandler from helix with the exisint ExponentialRetry code here, you'll get the behaviors you want with only 2-3 lines of code.

Reply:
I'll port the RetryAfterHandler to arcade-services in a separate PR

Reply:
The RetryAfterHandler has been ported and is being used now